### ISSUE TITLE
Member data for State and District is confusing

### ISSUE THREAD

**Issue by tomburge** [LibraryOfCongress/api.congress.gov#277]
I created a gist to showcase the following issue: https://gist.github.com/tomburge/d5b5f238eb5b51c3d14f222e6a744040

In the example1.json, this endpoint was used /member/{stateCode}/{district}.

In the example2.json, this endpoint was used /member/{bioguideId}.

In example1.json the query was for TX district 15. This returned 5 records.

3 had district 15, 1 had district 34, and 1 had no district.

In the example2.json, the member has 4 terms of which 3 are for district 15 and the current one is district 34.

It would be better (and less confusing) to have the district attribute inside of each term instead of a base attribute.

This would help with members who switched districts and for members who were in the House but then in the Senate.

**Comment by tomburge** [comment]
The workaround makes sense if you are only concerned about current members. I didn't include the tests, but I ran into some issues with existing districts like Texas districts 1-10 with specific Congresses sometimes reporting no results and I have to wonder if no results were returned because a member was included on a response for another district. It was weird.

**Comment by apreiter18** [comment]
@tomburge - yes, that is strange behavior, and I would like to investigate further. If you can recall any calls (I just tried Texas districts 1-10 for the 118th Congress and got results so maybe a previous Congress is the culprit?), please send them my way so I can take a look. I will keep digging in the meantime, too, to see if I can replicate this. Thanks for reporting this and following up!

**Comment by tomburge** [comment]
There are three I just checked:

Has Results:
Congress 118
District 3
State TX

No Results:
Congress 100
District 3
State TX

Has Results:
Congress 118
District 4
State TX

No Results:
Congress 101
District 4
State TX

Has Results:
Congress 118
District 7
State TX

No Results:
Congress 105
District 7
State TX

**Comment by apreiter18** [comment]
Yes, looking at these examples redistricting is the culprit for these three, @tomburge - thank you for sharing these. I am going to have discussions on our end about this to see if there are particular element(s) we can add to make this a bit clearer. Thanks, again, for reporting this.

**Comment by tomburge** [comment]
I just looped through Congresses 97-118 and I got this:

No member found for Congress: 97, District: 10, State: TX
No member found for Congress: 98, District: 10, State: TX
No member found for Congress: 99, District: 10, State: TX
No member found for Congress: 100, District: 10, State: TX
No member found for Congress: 101, District: 10, State: TX
No member found for Congress: 102, District: 10, State: TX
No member found for Congress: 103, District: 10, State: TX
Congress: 104, District: 10, State: TX
Member Name: Doggett, Lloyd
Member ID: D000399
Member Party: Democratic
Congress: 105, District: 10, State: TX
Member Name: Doggett, Lloyd
Member ID: D000399
Member Party: Democratic
Congress: 106, District: 10, State: TX
Member Name: Doggett, Lloyd
Member ID: D000399
Member Party: Democratic
Congress: 107, District: 10, State: TX
Member Name: Doggett, Lloyd
Member ID: D000399
Member Party: Democratic
Congress: 108, District: 10, State: TX
Member Name: Doggett, Lloyd
Member ID: D000399
Member Party: Democratic
Congress: 109, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 110, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 111, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 112, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 113, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 114, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 115, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 116, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 117, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican
Congress: 118, District: 10, State: TX
Member Name: McCaul, Michael T.
Member ID: M001157
Member Party: Republican

Looking at the Congress.gov site for members, I see this as well:

![image](https://github.com/user-attachments/assets/b37621ff-1fd8-42c1-be74-97884e5eac43)

The counts decrease farther back you go so it makes me think there may be an issue with the historical data or how it is presented.

**Comment by apreiter18** [comment]
@tomburge - you are quite correct, when you deselect "118" you will see the numbers increase. Currently, when "118" is selected it is showing members who are in the 118th Congress AND those who were in previous Congresses, as well. See below:
![image](https://github.com/user-attachments/assets/52a43d77-01ff-4630-bf76-a3e87f76f340)

One possible solution, and something I intend to update our documentation with, is to add a "currentMember=False" filter to your call (e.g., https://api.congress.gov/v3/member/congress/97/TX/10?currentMember=False&api_key=INSERT_KEY).

**Comment by tomburge** [comment]
I just validated with the same loop and adding currentMember = False fixes the issue.

Is it possible to add a default value of currentMember = False to the API to this call: GET /member/congress/:congress/:stateCode/:district? 

I would think if the API is being called with a specific Congress that isn't the current one then the response would be expected to be populated with the members of that specific Congress whether they are current or not.

**Comment by apreiter18** [comment]
@tomburge - Yes, we can update the documentation and API value to account for this. Thanks!
